src/state/Maps/actions.js

```diff
@@ -71,22 +99,71 @@ function layerUse(componentId, activeKeys, layer, spatialFilter) {
 
         // It converts modifiers from metadataKeys: ["A", "B"] to metadataKey: {in: ["A", "B"]}
         const modifiersForRequest = commonHelpers.convertModifiersToRequestFriendlyFormat(modifiers);
-
         if (layerTemplateKey || areaTreeLevelKey) {
+            let mergedFilter = {};
+            if(areaTreeLevelKey) {
+                mergedFilter = {...modifiersForRequest, areaTreeLevelKey};
+            }
+
+            if(layerTemplateKey) {
+                mergedFilter = {...modifiersForRequest, layerTemplateKey};
+            }
+
+
+            if(layerTemplateKey) {
+                const spatialDataSource = Select.data.spatialDataSources.getByFilteredIndex(state, mergedFilter, null);
```

Comment:
@vdubr Is there always one dataSource per merged filter?

Reply:
IMHO, there could be more spatial data sources linked to the same mergedFilter. For instace, if you have multiple shapefiles (e.g. one per country) and want to have one layer for whole Europe

Reply:
Fixed